local-cluster/src/Hydra/LogFilter.hs

```diff
@@ -0,0 +1,82 @@
+-- | Utility functions to filter and simplify raw logs.
+module Hydra.LogFilter where
+
+import Control.Lens (Traversal', at, (.~), (?~), (^..), (^?))
+import Data.Aeson (Value (Array, Null))
+import Data.Aeson.Lens (key, values, _Object)
+import Hydra.Prelude
+
+filterLog :: Value -> Maybe Value
+filterLog entry = do
+  guard (entry ^? key "message" . key "tag" == Just "Node")
+  node <- entry ^? key "message" . key "node"
+  result <- case node ^? key "tag" of
+    Just "ProcessingEvent" -> do
+      case (node ^? networkMessage . key "tag") <|> (node ^? clientEvent . key "tag") of
+        Just "ReqTx" -> processReqTx node
+        Just "NewTx" -> replaceTransactionWithTxId node clientEvent
+        Just "ReqSn" -> replaceTransactionsWithTxIds node networkMessage
+        _ -> pure node
+    Just "ProcessedEvent" -> do
+      case (node ^? networkMessage . key "tag") <|> (node ^? clientEvent . key "tag") of
+        Just "ReqTx" -> processReqTx node
+        Just "NewTx" -> replaceTransactionWithTxId node clientEvent
+        Just "ReqSn" -> replaceTransactionsWithTxIds node networkMessage
+        _ -> pure node
+    Just "ProcessingEffect" -> do
+      case node ^? clientEffect . key "tag" <|> node ^? networkEffect . key "tag" of
+        Just "SnapshotConfirmed" -> processSnapshotConfirmed node
+        Just "Committed" -> replaceUtxoWithNull node
+        Just "TxSeen" -> replaceTransactionWithTxId node clientEffect
+        Just "TxInvalid" -> replaceTransactionWithTxId node clientEffect >>= replaceUtxoWithNull
+        Just "TxValid" -> replaceTransactionWithTxId node clientEffect
+        Just "ReqTx" -> replaceTransactionWithTxId node networkEffect
+        Just "ReqSn" -> replaceTransactionsWithTxIds node networkEffect
+        _ -> pure node
+    Just "ProcessedEffect" -> do
```

Comment:
So far, I've rarely been interested in both `ProcessingXXX` and `ProcessedXXX`. I usually find them redundant in terms of information. Perhaps we could actually assign effects a short identifier and simply refer to them in the `ProcessedXXX` kind of events to avoid duplicating information and bloating logs even more.

Just a thought...

Reply:
100% agree. This program is more a tactical band-aid at the moment, although I think in the long run it's a good strategy to have specific log handlers for specific purposes. I also think we could reduce the footprint of our logs by referencing events and effects instead of repeating them, and improve the sturcture of messages.